src/services/github.service.ts

```diff
@@ -21,111 +40,101 @@ export class GithubService {
     const url: string = this.gitHubEndpoints.getUserUrl(username);
 
-    const result = await this.httpService
-      .get(url, {
-        headers: headers,
-      })
-      .toPromise()
-      .catch((err) => {
+    return this.githubApiClientService.get<GithubUser>(url, headers).pipe(
+      catchError((err) => {
         if (err.response && err.response.status === HttpStatus.NOT_FOUND) {
           throw new NotFoundException({
             status: HttpStatus.NOT_FOUND,
             message: 'GitHub user not found',
           });
         }
         throw err;
-      });
-
-    const user: User = {
-      login: result.data.login,
-      isOrg: result.data.type === UserTypeEnum.Organization,
-    };
-
-    return user;
+      }),
+      map((user) => {
+        return {
+          login: user.login,
+          isOrg: user.type === UserTypeEnum.Organization,
+        };
+      }),
+    );
   }
 
-  public async getNotForkRepos(
+  public getNotForkRepos(
     user: User,
     headers: HeadersForGit,
-  ): Promise<Repository[]> {
+  ): Observable<Repository[]> {
     const url: string = user.isOrg
       ? this.gitHubEndpoints.getOrgReposListUrl(user.login)
       : this.gitHubEndpoints.getUserReposListUrl(user.login);
 
-    const result = await this.httpService
-      .get(url, {
-        headers: headers,
-      })
-      .toPromise();
-
-    return result.data
-      .filter((repo) => !repo.fork)
-      .map((repo) => {
-        return <Repository>{
-          repository_name: repo.name,
-          owner_login: repo.owner.login,
-          branches: [],
-        };
-      });
+    return this.githubApiClientService
+      .get<GithubRepository[]>(url, headers)
+      .pipe(
+        map((repos) => repos.filter((repo) => !repo.fork)),
+        map((repos) =>
+          repos.map((repo) => {
+            return {
+              repository_name: repo.name,
+              owner_login: repo.owner.login,
+              branches: [],
+            };
+          }),
+        ),
+      );
   }
 
-  public async getBranches(
+  public getBranches(
     user: User,
     repo: Repository,
     headers: HeadersForGit,
-  ): Promise<Branch[]> {
+  ): Observable<Branch[]> {
     const url: string = this.gitHubEndpoints.getBranchesUrl(
       user.login,
       repo.repository_name,
     );
 
-    const result = await this.httpService
-      .get(url, {
-        headers: headers,
-      })
-      .toPromise();
-
-    return result.data.map((branch) => {
-      return <Branch>{
-        name: branch.name,
-        sha: branch.commit.sha,
-      };
-    });
+    return this.githubApiClientService.get<GithubBranch[]>(url, headers).pipe(
+      map((branches) =>
+        branches.map((branch) => {
+          return {
+            name: branch.name,
+            sha: branch.commit.sha,
+          };
+        }),
+      ),
+    );
   }
 
-  public async setBranchesToRepos(
+  public setBranchesToRepos(
```

Comment:
That does not look right, mutating state is Your enemy. This method should simply return Observable<Branch[]> and it should be called from method that constructs repos